hpack.go

```diff
@@ -392,6 +395,9 @@ func appendInt(dst []byte, bits uint8, index uint64) []byte {
 func readString(dst, b []byte) ([]byte, []byte, error) {
 	var n uint64
 	var err error
+	if len(b) < 1 {
```

Comment:
Can you just do `len(b) == 0`? `len` should always return >=0, otherwise something is really broken and the program should panic (in my opinion)

Reply:
For me,  we should never panic, panic just breaks the server. 
TodayI can make it panics if I send something invalid from my client and I think it's a problem.
May be we should handle it before calling this method but it seems more complicated for now.

Reply:
I mean, in such cases in which calling `len` returns a negative number, I think we should panic. The problem will come up sooner or later. But ok, if you prefer to not panic is ok.